client/src/questions-n-answers/allClicks.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import propTypes from 'prop-types';
+class AllClicks extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      loadAnswerState: false,
+
+
+    }
+    this.moreAnsweredQorLoadMoreAnswers = this.moreAnsweredQorLoadMoreAnswers.bind(this)
+  }
+  moreAnsweredQorLoadMoreAnswers(e, currentAnswerShowState) {
+    if (e.target.className === 'loadMoreAnswersButton') {
+      this.setState({
+        loadAnswerState: !this.state.loadAnswerState
+      })
+    }
+
+  }
+  render () {
+    const renderProps = {
+      loadAnswerState: this.state.loadAnswerState,
+      loadMoreAnsOrQ: this.moreAnsweredQorLoadMoreAnswers
+    }
+
+    return typeof this.props.children === 'function'
+    ? this.props.children(renderProps)
```

Comment:
Don't totally understand what is happening here. Is it like a recursive call to iterate through all children?